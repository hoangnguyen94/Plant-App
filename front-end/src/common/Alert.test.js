import React from 'react';
import { render } from '@testing-library/react';
import Alert from './Alert';

describe('Alert component', () => {
  test('renders the alert component with messages', () => {
    const messages = ['Error 1', 'Error 2'];
    const { container } = render(<Alert type="danger" messages={messages} />);
    
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders the alert component without messages', () => {
    const { container } = render(<Alert type="danger" messages={[]} />);
    
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders the alert component with a different type', () => {
    const messages = ['Warning'];
    const { container } = render(<Alert type="warning" messages={messages} />);
    
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders the alert component with multiple messages', () => {
    const messages = ['Message 1', 'Message 2', 'Message 3'];
    const { container } = render(<Alert type="info" messages={messages} />);
    
    expect(container.firstChild).toMatchSnapshot();
  });
});
