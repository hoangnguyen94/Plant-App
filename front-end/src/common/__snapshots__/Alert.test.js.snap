// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders the alert component with a different type 1`] = `
<div
  class="alert alert-warning"
  role="alert"
>
  <p
    class="mb-0 small"
  >
    Warning
  </p>
</div>
`;

exports[`Alert component renders the alert component with messages 1`] = `
<div
  class="alert alert-danger"
  role="alert"
>
  <p
    class="mb-0 small"
  >
    Error 1
  </p>
  <p
    class="mb-0 small"
  >
    Error 2
  </p>
</div>
`;

exports[`Alert component renders the alert component with multiple messages 1`] = `
<div
  class="alert alert-info"
  role="alert"
>
  <p
    class="mb-0 small"
  >
    Message 1
  </p>
  <p
    class="mb-0 small"
  >
    Message 2
  </p>
  <p
    class="mb-0 small"
  >
    Message 3
  </p>
</div>
`;

exports[`Alert component renders the alert component without messages 1`] = `
<div
  class="alert alert-danger"
  role="alert"
/>
`;
